### PR TITLE
fix issue #12657

### PR DESCRIPTION
**Pull request by redjumper**
修复issue #12657
pr https://github.com/siyuan-note/siyuan/pull/12596 对uid的判断有问题，还有漏了/siyuan/workspace的文件权限

我这边测试没问题

### CODE REVIEW

**Comment by 88250** [comment]
@Macavity Please review it

**Comment by redjumper** [comment]
如果指定了workspace，workspace的所有者权限也得改

**Comment by 88250** [comment]
> 如果指定了workspace，相应地得改workspace的所有者权限，而不是/siyuan/workspace

这个 PR 中没有实现对吗？只是写死了 /siyuan/workspace 路径？

**Comment by redjumper** [comment]
> 这个 PR 中没有实现对吗？只是写死了 /siyuan/workspace 路径？

没有实现，看Macavity的pr才看到有workspace参数。我修改的话暂时只想到引入env，如果这样修改，以前指定了workspace参数的得修改运行命令。等看看其它人有什么优雅的实现吧

**Review by Macavity — APPROVED** [review]
Looks good to me.

**Comment by redjumper** [comment]
@Macavity I believe that the ownership of the workspace should also change if the workspace is specified. This pr does not implement this.Do you have any good method?

**Comment by redjumper** [comment]
> > 如果指定了workspace，相应地得改workspace的所有者权限，而不是/siyuan/workspace
> 
> 这个 PR 中没有实现对吗？只是写死了 /siyuan/workspace 路径？

已经实现

File: kernel/entrypoint.sh [hunk 1]
```diff
@@ -27,12 +28,22 @@ else
...
+        *) ARGS="$ARGS $1"; shift ;;
+    esac
+done
+
+# Change ownership of relevant directories, including the workspace directory
+echo "Adjusting ownership of /opt/siyuan, /home/siyuan/, and ${WORKSPACE_DIR}"
 chown -R "${PUID}:${PGID}" /opt/siyuan
 chown -R "${PUID}:${PGID}" /home/siyuan/
-chown -R "${PUID}:${PGID}" /siyuan/workspace
+chown -R "${PUID}:${PGID}" "${WORKSPACE_DIR}"
```
Comment (Macavity) [inline kernel/entrypoint.sh:45]: That's a good idea 👍

**Review by Macavity — APPROVED** [review]
Thank you for adjusting the original changes and allow to adjust the workspace directory ownership as well 👍

**Comment by 88250** [comment]
Thanks to everyone for discussing and solving this issue, we merged it.

**Pull request merged**